.ci/lib/stage-test-apps-sgx.jenkinsfile

```diff
@@ -0,0 +1,184 @@
+stage('test') {
```

Comment:
Can we read a parameter which defines a set of tests to run? This is when you don't want to run everything but a few subset of tests.

We should implement some pytest interface or some runner which houses all the test code and you just provide the list of tests to run from the pipeline instead of having the test code as part of the pipeline. This will also provide result verification capabilities which I don't see right now. In future we can have this runner to house the performance tests too.

Same comment for all the test stage jenkinsfiles.